### DISCUSSION TITLE
Wireless Channel Model Simulation with Sionna

### DISCUSSION THREAD

**Discussion by KevalPankhaniya** [NVlabs/sionna#28] [General]
Hello All,

I want to really contribute to sionna. Because it is helping me a lot for my research. I think now it is time to also think about contribution. actually, I wanted to use winner 2 model for my research and got to know that it is available publicly on matlab. but matlab is not affordable for every researcher. I think if it will be available with sionna, it would be very nice. So, I just messaged to Jakob about this idea of integration of winner 2 with sionna. 

Yesterday I had a chat with Jakob on linked in regarding same. He appreciated and also suggested that it is good if it can be combined with 38.901 channel models. So, I am starting this initial discussion here. Kindly suggest and provide the direction about the development.

Kindly also go through the following links for reference. 

https://in.mathworks.com/help/comm/ug/winner-ii-channel.html
https://www.researchgate.net/publication/259900906_IST-4-027756_WINNER_II_D112_v12_WINNER_II_channel_models
[tr_138901v160100p.pdf](https://github.com/NVlabs/sionna/files/8886172/tr_138901v160100p.pdf)

**Comment by jhoydis** [comment]
Hi,

Thanks for this suggestion. I have had a detailed look at the WINNER II specification and have a couple of questions/comments:

1) It seems that 38.901 is largely based on WINNER II and extends it in many ways. Which feature of WINNER II are you missing in the 38.901 models?

2) Given that the procedures for the generation of large- and small-scale parameters are identical, the implementation of WINNER II would essentially boil down to adding new JSON configuration files for the different Scenarios A1, A2, ..., as well as dedicated functions for the pathloss and LOS probability computations. This could be very easily done. Also the CDL models could be very easily integrated.

3) Rather than adding the WINNER II scenarios, it might be more useful to add the 38.901 InH and InF models. So, I would probably prioritise this.

4) We would probably not implement the bad urban channels (B2, C3) as this requires placing scatterers within a cell. This could be done, but would probably require substantial work for little added value to the users.

5) WINNER II antenna arrays and patterns could be added.

6) Keep in mind that Sionna does not support specific trajectory design with different segments, LOS/NLOS transitions, etc. It is currently not planned to add this feature.

**Reply by KevalPankhaniya** [reply]
1. It seems that 38.901 is largely based on WINNER II and extends it in many ways. Which feature of WINNER II are you missing in the 38.901 models?
----> Going through the specification and also the winner 2 implementation. I will post here about the either the enhancement or the missing feature that we can include. 

2. Given that the procedures for the generation of large- and small-scale parameters are identical, the implementation of WINNER II would essentially boil down to adding new JSON configuration files for the different Scenarios A1, A2, ..., as well as dedicated functions for the pathloss and LOS probability computations. This could be very easily done. Also the CDL models could be very easily integrated.
-----> I think that is great and we can always start with the easy steps first.


3. Rather than adding the WINNER II scenarios, it might be more useful to add the 38.901 InH and InF models. So, I would probably prioritise this.
-----> Let me go through those models. 

4. We would probably not implement the bad urban channels (B2, C3) as this requires placing scatterers within a cell. This could be done, but would probably require substantial work for little added value to the users.
-----> This can be done in the future then. 

5. WINNER II antenna arrays and patterns could be added.
-----> I needs to check this. 

6. Keep in mind that Sionna does not support specific trajectory design with different segments, LOS/NLOS transitions, etc. It is currently not planned to add this feature.
-----> we might include that in future. it requires huge efforts from the community. 


I think we can start with the easiest implementation, first. then we can plan for future things. 

So, do we needs to make any time line for the same ? 

Kindly give a few words to kick start the activity. I think it would be better if we discuss and plan first thoroughly before starting implementation.

**Comment by jhoydis** [comment]
After some internal discussions, we believe that it would be best to start by adding the 38.901 InF and InO models. Let us know if you require some help to get started.

**Reply by KevalPankhaniya** [reply]
Hi Jakob, implementation will be in C and then C library we can link in python, that is fine, or python only is required for implementation ?Even I am going through the https://quadriga-channel-model.de/

**Reply by jhoydis** [reply]
Sionna is entirely written in TensorFlow. Any contribution must therefore be also in TensorFlow. This is important as the channel model must be part of the computational graph. Otherwise, we could simply rely on existing channel simulators.
